### PR TITLE
add basic healthcheck feature #105

### PR DESCRIPTION
**Pull request by Hechamon**
Summary of the changes (in less than 80 chars)

 * Added the most basic healthz endpoint

Addresses https://github.com/bagetter/BaGet/issues/105
I tried adding the database provider health checks at the same time, however the way the registration of those via the IProvider interface made it a lot more difficult than I have time for right now.

### CODE REVIEW

**Comment by seriouz** [comment]
@Hechamon issue 123 does not exist. Which do you refer to?

**Comment by Hechamon** [comment]
> @Hechamon issue 123 does not exist. Which do you refer to?

105, sorry :-)

**Comment by seriouz** [comment]
Can you make the slug/path of the health check overwriteable via a config `HealthCheckPath`?
Additionally a mention somewhere in the docs that we now have a health check and its default value would be good.
Thank you!

**Comment by FroggieFrog** [comment]
Call me oldschool, but I prefer a more common name for routes. So I prefer the more common `/health` route name.
In rare cases I also came across `/hc`.

If I find some time the next days and if you would like, I could try to extend/adjust this PR. Because you said you are short of time.

File: src/BaGetter/Startup.cs [hunk 1]
```diff
@@ -89,5 +91,7 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
 
             baget.MapEndpoints(endpoints);
         });
+
+        app.UseHealthChecks("/healthz");
```
Comment (Regenhardt) [inline src/BaGetter/Startup.cs:95]:
```suggestion
        app.UseHealthChecks("/health");
```

**Review by Regenhardt — CHANGES_REQUESTED** [review]

File: src/BaGetter/Startup.cs (hunk 1 above)
Comment (Regenhardt) [reply src/BaGetter/Startup.cs:95]: I recently learning of this very nice github feature to request specific changes and directly commit it with a click. Please use a common name for the health check endpoint, as this is complicated enough to find without an OpenAPI endpoint.

Comment (seriouz) [reply src/BaGetter/Startup.cs:95]: I would more like an overwriteable endpoint, but `health` as default.

Comment (Hechamon) [reply src/BaGetter/Startup.cs:95]: I will change to be configurable with a default of /health, as @seriouz suggested.
The name healthz is very common in containerized applications, which is why I am used to it.
[See this stackoverflow question](https://stackoverflow.com/questions/43380939/where-does-the-convention-of-using-healthz-for-application-health-checks-come-f)

**Comment by Hechamon** [comment]
I have pushed a new commit to make the endpoint configurable with a default of /health

**Review by seriouz — APPROVED** [review]
Thank you very much!

File: docs/docs/configuration.md [hunk 2]
```diff
@@ -170,6 +170,25 @@ If not specified, the `MaxRequestBodySize` in BaGetter defaults to 250MB (262144
 }
 ```
 
+## Health Endpoint
+
+When running within a containerized environment like Kubernetes, a basic health endpoint is exposed at `/health` that returns 200 OK and the text "Healthy" when running.
+
+This path is configurable if needed:
+
```
Comment (FroggieFrog) [inline docs/docs/configuration.md:178]: Please only one single new line

File: docs/docs/configuration.md [hunk 3]
```diff
@@ -180,6 +199,7 @@ Mostly useful when running containerised (e.g. using Docker, Podman, Kubernetes,
 
 This allows for sensitive values to be provided individually to the application, typically by bind-mounting files.
 
+
```
Comment (FroggieFrog) [inline docs/docs/configuration.md:202]: Please only one single new line

**Review by FroggieFrog — APPROVED** [review]
Good addition to the project.

**Review by Regenhardt — APPROVED** [review]
LGTM, thanks!

**Pull request merged**